### PR TITLE
rgw: disable sync squash for versioned objects

### PR DESCRIPTION
**Pull request by yehudasa**
and also modify mtime reported on unlink_instance operation.

Fixes: http://tracker.ceph.com/issues/18939

### CODE REVIEW

File: src/rgw/rgw_data_sync.cc [hunk 1]
```diff
@@ -2530,11 +2534,12 @@ int RGWBucketShardIncrementalSyncCR::operate()
           marker_tracker.try_update_high_marker(cur_id, 0, entry->timestamp);
           continue;
         }
-        if (make_pair<>(entry->timestamp, entry->op) != squash_map[make_pair(entry->object, entry->instance)]) {
+        if (entry->instance.empty() &&
+            make_pair<>(entry->timestamp, entry->op) != squash_map[make_pair(entry->object, entry->instance)]) {
```
Comment (cbodley) [inline src/rgw/rgw_data_sync.cc]: can we revert the commit ce7d00ac1642d84c1d9111156a544c37801c5adf that added instance to the `squash_map` keys now that we're not using it?

Comment (yehudasa) [inline src/rgw/rgw_data_sync.cc]: @cbodley I'm not extremely happy about this fix, would rather understand the exact root cause before. The mtime issue was a problem, but I think there's another issue coming into play (I think it's del squashing unlink_instance), but still need to verify.

Comment (cbodley) [inline src/rgw/rgw_data_sync.cc]: that commit removed a special case related to `OP_DEL`, maybe that's relevant to this issue

**Comment by cbodley** [comment]
also `Fixes: http://tracker.ceph.com/issues/18885`

**Comment by yehudasa** [comment]
@mattbenjamin @oritwas rebased

**Comment by cbodley** [comment]
pulled the mtime change into a separate pr https://github.com/ceph/ceph/pull/20016

**Comment by yuriw** [comment]
@yehudasa pls rebase

--- pr 13426 --- pulling https://github.com/yehudasa/ceph.git branch wip-18939
remote: Counting objects: 11, done.
remote: Total 11 (delta 9), reused 9 (delta 9), pack-reused 2
Unpacking objects: 100% (11/11), done.
From https://github.com/yehudasa/ceph
 * branch            wip-18939  -> FETCH_HEAD
Auto-merging src/rgw/rgw_data_sync.cc
CONFLICT (file/directory): There is a directory with name src/dmclock in 9b10bcee74c68af00e823452409db698c9ab7a8a. Adding src/dmclock as src/dmclock~HEAD
Adding qa/suites/rgw/verify/frontend
Adding qa/suites/rgw/tempest/frontend
Adding qa/suites/rgw/multisite/frontend
Adding qa/suites/rbd/mirror/cluster
Adding qa/suites/rbd/mirror/base
Adding qa/suites/ceph-deploy/basic/distros/ubuntu_latest.yaml
Adding qa/suites/ceph-deploy/basic/distros/centos_latest.yaml
Adding qa/suites/ceph-deploy/basic/distros/.qa
Automatic merge failed; fix conflicts and then commit the result.
Traceback (most recent call last):
  File "/home/yuriw/wip_master/src/script/build-integration-branch", line 62, in <module>
    assert not r
AssertionError

**Pull request closed without merging**